### PR TITLE
Calculated and saved more necessary phonon calculation results

### PR DESCRIPTION
**Pull request by rul048**
Our codes used to only allow the calculation and return of thermal properties, ignoring band structure, force constants, etc., which limited further investigation, like dynamic stability, anharmonicity, etc.

In this update, we added several features. Phonon object was saved by default because it contained a lot of information, and the band structure, density of states, etc. can be reconstructed from this file. We also allowed users to determine whether to calculate and save band structure, density of states and force constants. (Default to False to ensure consistency with previous code and save storage space)

## Summary

Major changes:

- feature 1: ...
- fix 1: ...

## Todos

If this is work in progress, what else needs to be done?

- feature 2: ...
- fix 2:

## Checklist

- [ ] Google format doc strings added. Check with `ruff`.
- [ ] Type annotations included. Check with `mypy`.
- [ ] Tests added for new features/fixes.
- [ ] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

### CODE REVIEW

**Review by janosh — CHANGES_REQUESTED** [review]
thanks, very sensible change!

File: tests/test_phonon.py [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+    # change dir to tmp_path
+    phonon_yaml = tmp_path / "phonon.yaml"
     phonon_calc = PhononCalc(
-        M3GNetCalc, supercell_matrix=((2, 0, 0), (0, 2, 0), (0, 0, 2)), fmax=0.1, t_step=50, t_max=1000
+        M3GNetCalc,
+        supercell_matrix=((2, 0, 0), (0, 2, 0), (0, 0, 2)),
+        fmax=0.1,
+        t_step=50,
+        t_max=1000,
+        write_phonon=phonon_yaml,
```
Comment (janosh) [inline tests/test_phonon.py:31]: i added a test for `write_phonon`. we need to cover the other new keywords as well.

**Comment by shyuep** [comment]
Can we get this done soon?

**Comment by rul048** [comment]
> Can we get this done soon?

Thanks for your kind reminder, professor. Learned something new about pytest today.

**Comment by janosh** [comment]
i think we want to aim for more comprehensive tests than were merged here.

i just pushed more stringent `PhononCalc` tests in [df738a9](https://github.com/materialsvirtuallab/matcalc/commit/df738a98494b39f252c51a0df416296181c24b4e)

**Pull request merged**